### PR TITLE
Fix for #723

### PR DESCRIPTION
**Pull request by ili**
Fix for #723 
Firebird creates trigger too, but it does not support schemas.

### CODE REVIEW

File: Tests/Linq/DataProvider/OracleTests.cs [hunk 1]
```diff
@@ -1838,5 +1838,32 @@ public void Issue539(string context)
...
+		public void Issue723(string context)
+		{
+			using (var db = new TestDataConnection(context))
+			{
+				try
+				{
+					db.Execute("GRANT CREATE ANY TRIGGER TO TestUser");
+					db.Execute("CREATE USER Issue723Schema IDENTIFIED BY password");
+					db.CreateTable<Issue723Table>(schemaName: "Issue723Schema");
+				}
```
Comment (MaceWindu) [inline Tests/Linq/DataProvider/OracleTests.cs]: Could you add assert that LastQuery contains proper name?

Comment (ili) [inline Tests/Linq/DataProvider/OracleTests.cs]: You mean `Assert.That(db.LastQuery.Contains("Issue723Schema.Issue723Table"))`?

Comment (MaceWindu) [inline Tests/Linq/DataProvider/OracleTests.cs]: Yes, or by quering schema information from server

**Comment by Lexey** [comment]
It looks like it fixes the missing schema for the CREATE TRIGGER. But, as it has been said in #723 the same problem exists for the CREATE SEQUENCE.

**Comment by ili** [comment]
@Lexey `CREATE SEQUENCE` does not refers table.. or you do mean we should change sequence name?

**Comment by Lexey** [comment]
@ili Yes, I mean that the sequence name should the prefixed with the schema name.

File: Source/DataExtensions.cs [hunk 2]
```diff
@@ -266,33 +266,33 @@ public static int Delete<T>([NotNull] this IDataContext dataContext, T obj)
 		public static ITable<T> CreateTable<T>([NotNull] this IDataContextInfo dataContextInfo,
 			string         tableName       = null,
 			string         databaseName    = null,
-			string         ownerName       = null,
+			string         schemaName      = null,
```
Comment (sdanyliv) [inline Source/DataExtensions.cs]: Why you have removed owner name?
[Owner and Schma](http://www.sqlteam.com/article/understanding-the-difference-between-owners-and-schemas-in-sql-server)

Comment (Lexey) [inline Source/DataExtensions.cs]: I have requested this rename. It is really the schema name. ownerName just confuses a potential user of the method.

**Comment by ili** [comment]
@Lexey @sdanyliv @MaceWindu can anybody check last changes with oracle? I do have strange problems with DBHost VM (it does not see network) so I can't do it myself :(

**Comment by MaceWindu** [comment]
@ili I will check it today

**Comment by MaceWindu** [comment]
ok, managed to test it finally. I think this is not what @Lexey  expected:
```sql
CREATE SEQUENCE SIDENTITY_Issue723SchemaIssue723Table
Message  : ORA-00972: identifier is too long
```
I think it should be something like
```sql
CREATE SEQUENCE Issue723Schema.SIDENTITY_Issue723Table
```

BTW, I've updated test a bit to work more stable:
```
[Test, OracleDataContext]
		public void Issue723Test(string context)
		{
			using (var db = new TestDataConnection(context))
			{
				var currentUser = db.Execute<string>("SELECT user FROM dual");
				db.Execute("GRANT CREATE ANY TRIGGER TO " + currentUser);

				db.Execute("CREATE USER Issue723Schema IDENTIFIED BY password");

				try
				{
					var tableSpace = db.Execute<string>("SELECT default_tablespace FROM sys.dba_users WHERE username = 'ISSUE723SCHEMA'");
					db.Execute("ALTER USER Issue723Schema quota unlimited on {0}".Args(tableSpace));

					db.CreateTable<Issue723Table>(schemaName: "Issue723Schema");

					try
					{
						for (var i = 1; i < 3; i++)
						{
							var id = Convert.ToInt32(db.InsertWithIdentity(new Issue723Table() { StringValue = i.ToString() }));
							Assert.AreEqual(i, id);
						}
						Assert.That(db.LastQuery.Contains("Issue723Schema.Issue723Table"));
					}
					finally
					{
						db.DropTable<Issue723Table>(schemaName: "Issue723Schema");
					}
				}
				finally
				{
					db.Execute("DROP USER Issue723Schema CASCADE");
				}
			}
		}
```

**Comment by Lexey** [comment]
@MaceWindu, you are right. That is not I have expected. :)
Even after an obvious bugfix of GetFullYabelName
```c#
return table.Owner != null ? table.Owner + '.' + table.PhysicalName : table.PhysicalName;
```
it produces wrong results. 
Really the name generation code should be like:
```c#
				var table = SelectQuery.CreateTable.Table;
				StringBuilder
					.Append("DROP TRIGGER ");
				if (table.Owner != null) StringBuilder.Append(table.Owner).Append('.');
				StringBuilder
					.Append("TIDENTITY_")
					.Append(table.PhysicalName)
					.AppendLine();
```

**Comment by ili** [comment]
@Lexey & @MaceWindu what about now? :)

File: Source/DataProvider/Oracle/OracleSqlBuilder.cs [hunk 3]
```diff
@@ -288,6 +288,11 @@ public override int CommandCount(SelectQuery selectQuery)
 			return base.CommandCount(selectQuery);
 		}
 
+		private static string GetFullTableName(SqlTable table)
```
Comment (MaceWindu) [inline Source/DataProvider/Oracle/OracleSqlBuilder.cs]: Looks like this method is not used anymore

File: Source/DataProvider/Oracle/OracleSqlBuilder.cs [hunk 4]
```diff
@@ -297,7 +302,9 @@ protected override void BuildDropTableStatement()
 			else
 			{
 				StringBuilder
-					.Append("DROP TRIGGER TIDENTITY_")
+					.Append("DROP TRIGGER ")
+					.Append(SelectQuery.CreateTable.Table.Owner)
+					.Append(".TIDENTITY_")
```
Comment (MaceWindu) [inline Source/DataProvider/Oracle/OracleSqlBuilder.cs]: It looks like without owner, FQN will start with dot. Will Oracle handle it properly?
I think we should add second test that perfroms same actions without schema name

File: Source/DataProvider/Oracle/OracleSqlBuilder.cs [hunk 5]
```diff
@@ -310,7 +317,9 @@ protected override void BuildCommand(int commandNumber)
 				if (commandNumber == 1)
 				{
 					StringBuilder
-						.Append("DROP SEQUENCE SIDENTITY_")
+						.Append("DROP SEQUENCE ")
+						.Append(SelectQuery.CreateTable.Table.Owner)
+						.Append(".SIDENTITY_")
```
Comment (MaceWindu) [inline Source/DataProvider/Oracle/OracleSqlBuilder.cs]: same question about schema-less case

File: Source/DataProvider/Oracle/OracleSqlBuilder.cs [hunk 6]
```diff
@@ -322,19 +331,26 @@ protected override void BuildCommand(int commandNumber)
...
 						.Append(SelectQuery.CreateTable.Table.PhysicalName)
 						.AppendLine();
 				}
 				else
 				{
 					StringBuilder
-						.AppendFormat("CREATE OR REPLACE TRIGGER  TIDENTITY_{0}", SelectQuery.CreateTable.Table.PhysicalName)
-						.AppendLine  ()
-						.AppendFormat("BEFORE INSERT ON {0} FOR EACH ROW", SelectQuery.CreateTable.Table.PhysicalName)
+						.AppendFormat("CREATE OR REPLACE TRIGGER {0}.TIDENTITY_{1}", SelectQuery.CreateTable.Table.Owner, SelectQuery.CreateTable.Table.PhysicalName)
```
Comment (MaceWindu) [inline Source/DataProvider/Oracle/OracleSqlBuilder.cs]: same question about schema-less case

File: Source/DataProvider/Oracle/OracleSqlBuilder.cs [hunk 7]
```diff
@@ -322,19 +331,26 @@ protected override void BuildCommand(int commandNumber)
...
+						.AppendFormat("BEFORE INSERT ON ");
+
+					BuildPhysicalTable(SelectQuery.CreateTable.Table, null);
+
+					StringBuilder
+						.AppendLine(" FOR EACH ROW")
 						.AppendLine  ()
 						.AppendLine  ("BEGIN")
-						.AppendFormat("\tSELECT SIDENTITY_{1}.NEXTVAL INTO :NEW.{0} FROM dual;", _identityField.PhysicalName, SelectQuery.CreateTable.Table.PhysicalName)
+						.AppendFormat("\tSELECT {2}.SIDENTITY_{1}.NEXTVAL INTO :NEW.{0} FROM dual;", _identityField.PhysicalName, SelectQuery.CreateTable.Table.PhysicalName, SelectQuery.CreateTable.Table.Owner)
```
Comment (MaceWindu) [inline Source/DataProvider/Oracle/OracleSqlBuilder.cs]: same question about schema-less case

File: Tests/Linq/DataProvider/OracleTests.cs [hunk 8]
```diff
@@ -1838,5 +1838,62 @@ public void Issue539(string context)
...
+				db.Execute("CREATE USER Issue723Schema IDENTIFIED BY password");
+
+				try
+				{
+
+					var tableSpace = db.Execute<string>("SELECT default_tablespace FROM sys.dba_users WHERE username = 'ISSUE723SCHEMA'");
+					db.Execute("ALTER USER Issue723Schema quota unlimited on {0}".Args(tableSpace));
+
+					db.CreateTable<Issue723Table>(schemaName: "Issue723Schema");
+					Assert.That(db.LastQuery.Contains("Issue723Schema.Issue723Table"));
```
Comment (MaceWindu) [inline Tests/Linq/DataProvider/OracleTests.cs]: move assert to try, so drop table will be called even if assert failed

File: Source/DataProvider/Oracle/OracleSqlBuilder.cs (hunk 4 above)
Comment (Lexey) [inline Source/DataProvider/Oracle/OracleSqlBuilder.cs]: AFAIR, it will fail in this case

**Comment by ili** [comment]
@Lexey can you make a review please? :)

File: Tests/Linq/DataProvider/OracleTests.cs [hunk 9]
```diff
@@ -1839,6 +1839,73 @@ public void Issue539(string context)
...
+
+				try
+				{
+
+					var tableSpace = db.Execute<string>("SELECT default_tablespace FROM sys.dba_users WHERE username = 'ISSUE723SCHEMA'");
+					db.Execute("ALTER USER Issue723Schema quota unlimited on {0}".Args(tableSpace));
+					try
+					{
+
+						db.CreateTable<Issue723Table>(schemaName: "Issue723Schema");
```
Comment (MaceWindu) [inline Tests/Linq/DataProvider/OracleTests.cs]: Let's move CreateTable before ```try``` so it will not try to delete it in finally when create failed

**Pull request merged**